frontend/public/components/overview/index.tsx

```diff
@@ -486,11 +521,45 @@ class OverviewDetails extends SafetyFirst<OverviewDetailsProps, OverviewDetailsS
     };
   }
 
+  getBuildsForResource(buildConfig) {
+    const {builds} = this.props;
+    return getOwnedResources(buildConfig, builds.data);
+  }
+
+  getBuildConfigsForResource(resource) {
+    const {buildConfigs} = this.props;
+    const nativeTriggers = _.get(resource, 'spec.triggers');
+    const annotatedTriggers = JSON.parse(_.get(resource, ['metadata','annotations', TRIGGERS_ANNOTATION], '[]'));
```

Comment:
We should catch exceptions parsing the JSON, or we white screen if someone edits the value to something bad.

```ts
const getAnnotatedTriggers = (obj: K8sResourceKind) => {
  const triggersJSON = getAnnotation(resource, TRIGGERS_ANNOTATION) || '[]';
  try {
    return JSON.parse(triggersJSON);
  } catch(e) {
    console.warn('Error parsing triggers annotation', e);
    return [];
  }
};
```

We should eventually pull this into a utility that we can use on workload details pages, but we can do that later.

Reply:
Yeah, I feel like a lot of the functions we have in the top of this file could be pulled out into utilities.